### PR TITLE
Allow selectively redacting diff content 

### PR DESCRIPTION
**Pull request by Oded-B**
## Description

Add a new component-level "disableDiff" configuration key
Read the configuration file before diff
Pass the the config the relevant function
Replace diff content with a relevant message

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [ ] I have read the [contributing guidelines](https://github.com/wayfair-incubator/telefonistka/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -184,20 +184,31 @@ toggleCommitStatus:
 This optional in-component configuration file allows overriding the general promotion configuration for a specific component.
 File location is `COMPONENT_PATH/telefonistka.yaml` (no leading dot in file name), so it could be:
 `workspace/reloader/telefonistka.yaml` or `env/prod/us-central1/c2/wf-kube-proxy-metrics-proxy/telefonistka.yaml`
-it includes only two optional configuration keys, `promotionTargetBlockList` and `promotionTargetAllowList`.
-Both are matched against the target component path using Golang regex engine.
+it includes these  optional configuration keys: `promotionTargetBlockList`,  `promotionTargetAllowList` and `disableArgoCDDiff`
+`promotionTargetBlockList` and `promotionTargetAllowLis  are matched against the target component path using Golang regex engine.
```
Comment (yzdann) [inline docs/installation.md:188]:
```suggestion
`promotionTargetBlockList` and `promotionTargetAllowList`  are matched against the target component path using Golang regex engine.
```

File: internal/pkg/argocd/argocd.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ type DiffResult struct {
 
 // Mostly copied from  https://github.com/argoproj/argo-cd/blob/4f6a8dce80f0accef7ed3b5510e178a6b398b331/cmd/argocd/commands/app.go#L1255C6-L1338
 // But instead of printing the diff to stdout, we return it as a string in a struct so we can format it in a nice PR comment.
```
Comment (yzdann) [inline internal/pkg/argocd/argocd.go:55]: Maybe we want to update this comment too as the function signature is also different now.

**Review by ashvarts — APPROVED** [review]
lgtm

File: internal/pkg/argocd/argocd.go [hunk 3]
```diff
@@ -53,7 +53,7 @@ type DiffResult struct {
...
-func generateArgocdAppDiff(ctx context.Context, app *argoappv1.Application, proj *argoappv1.AppProject, resources *application.ManagedResourcesResponse, argoSettings *settings.Settings, diffOptions *DifferenceOption) (foundDiffs bool, diffElements []DiffElement, err error) {
+func generateArgocdAppDiff(ctx context.Context, keepDiffData bool, app *argoappv1.Application, proj *argoappv1.AppProject, resources *application.ManagedResourcesResponse, argoSettings *settings.Settings, diffOptions *DifferenceOption) (foundDiffs bool, diffElements []DiffElement, err error) {
```
Comment (yzdann) [inline internal/pkg/argocd/argocd.go:56]: we can embed DifferenceOption, and add a new field called disable, then we don't need to pass an extra arg to this function, but i'm good with this either way.

File: internal/pkg/argocd/argocd.go [hunk 4]
```diff
@@ -126,7 +126,11 @@ func generateArgocdAppDiff(ctx context.Context, app *argoappv1.Application, proj
...
 
-			diffElement.Diff, err = diffLiveVsTargetObject(live, target)
+			if keepDiffData {
+				diffElement.Diff, err = diffLiveVsTargetObject(live, target)
+			} else {
+				diffElement.Diff = "✂️ ✂️  Redacted ✂️ ✂️ \nUnset component-level configuration key `disableArgoCDDiff` to see diff content."
+			}
 			if err != nil {
 				return false, nil, fmt.Errorf("Failed to diff live objects: %w", err)
 			}
```
Comment (yzdann) [inline internal/pkg/argocd/argocd.go:136]: not related to this pr but i think we already checked the error here -> https://github.com/commercetools/telefonistka/blob/e35f54e3bec2444c5a3433a650e47d3af886144b/internal/pkg/argocd/argocd.go#L102-L104

**Review by yzdann — APPROVED** [review]
I found naming a bit hard to understand.
We had `commentDiff` as bool, `componentsToDiff` as map of bool, but let's unblock the other work.

**Pull request merged**